libraries/aws_kms__key.rb

```diff
@@ -0,0 +1,92 @@
+class AwsKmsKey < Inspec.resource(1)
+  name 'aws_kms_key'
+  desc 'Verifies settings for an individual AWS KMS Key'
+  example "
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/key-id') do
+      it { should exist }
+    end
+  "
+
+  include AwsResourceMixin
+
+  attr_reader :key_id, :arn, :creation_date, :key_usage, :key_state, :description,
+              :deletion_date, :valid_to, :origin, :expiration_model, :key_manager
+
+  def to_s
+    "KMS Key #{@key_arn}"
+  end
+
+  def enabled?
+    @enabled
+  end
+
+  def rotation_enabled?
+    @rotation_enabled
+  end
+
+  def created_days_ago
+    ((Time.now - creation_date)/(24*60*60)).to_i unless creation_date.nil?
+  end
+
+  private
+
+  def validate_params(raw_params)
+    validated_params = check_resource_param_names(
+      raw_params: raw_params,
+      allowed_params: [:key_arn],
+      allowed_scalar_name: :key_arn,
+      allowed_scalar_type: String,
+    )
+
+    if validated_params.empty?
+      raise ArgumentError, "You must provide the parameter 'key_id' to aws_kms_key."
+    end
+
+    validated_params
+  end
+
+  def fetch_from_aws
+    backend = AwsKmsKey::BackendFactory.create
+
+    query = { key_id: @key_arn }
+    resp = backend.describe_key(query)
+
+    @exists = !resp.empty?
+    return unless @exists
+
+    @key = resp.key_metadata.to_h
```

Comment:
@key should be a local variable, not an instance variable.  Avoid inner objects.